core-blocks/quote/editor.scss

```diff
@@ -1,6 +1,10 @@
 .wp-block-quote {
 	margin: 0;
 
+	// compensate nested blocks for collapsing margins
+	padding-top: 1px;
```

Comment:
Is there a way to apply those styles to all blocks that use nesting? It might not scale well if we will have to do the same trick for all existing containers like `Columns` or `Pullquote`, soon `Cover Image` or `List`.

Reply:
Yes, mentioned in the PR: 

> This PR adds that to the blockquote block itself, but if we think the general approach in this PR has merit, then we should find a way to make this more generic. For example a block that has nested children, if it had a `has-children` classname, then we could simply add the padding to that.

So, if we think this approach has merit, what do you think of applying a `has-children` class to the parent block container? If we add that generic class, I can also make this code generic.

Reply:
Yes, this could be done with hooks in a similar way we add `data-align` to the block wrappers. Not sure if another class name or data attribute is better, but we could use such general solution 👍